.github/workflows/build-quesma-docker-image.yml

```diff
@@ -0,0 +1,88 @@
+# Reusable workflow that builds Quesma Docker image and either:
+# - pushes it to Docker Hub
+# - or exports it to .tar artifact
+
+name: Docker image build
+on:
+  workflow_call: # Can be reused from another workflow
+    inputs:
+      VERSION:
+        description: 'Version number to tag the image with (optional)'
+        default: nightly
+        required: false
+        type: string
+      PUSH:
+        description: 'Whether to push the image to the registry (optional)' # if not, we will export it to .tar artifact
+        default: false
+        required: false
+        type: boolean
+
+jobs:
+  build-quesma-docker-image:
+    strategy:
+      matrix:
+        module: [ "quesma" ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          cache-dependency-path: ${{ matrix.module }}/go.sum
+          go-version: '1.22'
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        if: ${{ inputs.PUSH }}
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PAT }}
+
+      - name: Set the build date
+        run: echo QUESMA_BUILD_DATE=$(git --no-pager log -1 --date=format:'%Y-%m-%d' --format="%ad")  >> $GITHUB_ENV
+
+      # If we don't push the image, we build it only for the host architecture and export it to .tar file
+      - name: Build and export
+        uses: docker/build-push-action@v6
+        if: ${{ !inputs.PUSH }}
+        with:
+          context: ${{ matrix.module }}/.
+          tags: |
+            quesma/quesma:${{ inputs.VERSION }}
+            quesma/quesma:nightly
+          push: false
+          outputs: type=docker,dest=/tmp/image.tar
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+        env:
+          DOCKER_BUILD_SUMMARY: false
+
+      # If we push the image, we build it for both amd64 and arm64 and don't export it
+      - name: Build and push
```

Comment:
The reason there's this duplication of "Build and export" and "Build and push" steps is that the first one doesn't build for arm64 in order to save time (arm64 build takes a lot of time on GitHub Actions, it's probably running emulated ARM on x86). But on the flip side, this preserves the exactly same behavior as before.

Reply:
I think this GitHub comment should land in the yaml 😉